README.md

```diff
@@ -17,6 +17,7 @@ This plugin currently supports Moodle:
 
 | Moodle version      | Branch               |
 | ------------------- | -------------------- |
+| Moodle 4.2 and up   | MOODLE_402_STABLE    |
```

Comment:
Are 4.0 and 4.1 not supported, if they are, which branch should be used? Seems to be a gap here. 

Could you check this please.

Reply:
Due to https://tracker.moodle.org/browse/MDL-76583 Moodle 4.0 and 4.1 are not supported. I have amended the readme commit accordingly. Even though the commit introduces the namespace I have also given this a test locally. Here are the results of that.

<details>
<summary>PHPUnit tests results with changes on Moodle 4.1.8 (Build: 20231222)</summary>

```
root@b0ba46b747f2:/var/www/site# vendor/bin/phpunit --testsuite search_elastic_testsuite
Moodle 4.1.8 (Build: 20231222), e591ddc4e8c21df9fb5369e388ef57ef07acbb9b
Php: 8.1.24, pgsql: 14.0 (Debian 14.0-1.pgdg110+1), OS: Linux 6.4.12-060412-generic x86_64
PHPUnit 9.5.28 by Sebastian Bergmann and contributors.

...............................................EE..............  63 / 115 ( 54%)
....................................................            115 / 115 (100%)

Time: 02:21.574, Memory: 78.50 MB

There were 2 errors:

1) search_elastic\externallib_test::test_external_search
Error: Class "core_external\external_api" not found

/var/www/site/search/engine/elastic/externallib.php:40
/var/www/site/search/engine/elastic/tests/externallib_test.php:103
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80

2) search_elastic\externallib_test::test_external_search_areas
Error: Class "search_elastic_external" not found

/var/www/site/search/engine/elastic/tests/externallib_test.php:172
/var/www/site/lib/phpunit/classes/advanced_testcase.php:80

ERRORS!
Tests: 115, Assertions: 219, Errors: 2.
```
</details>